clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1488,4 +1488,12 @@ public static String toLogDateTimeFormat(long timestamp) {
         return Instant.ofEpochMilli(timestamp).atZone(ZoneId.systemDefault()).format(dateTimeFormatter);
     }
 
+    /**
+     * Replace the given string suffix with the new suffix. If the string doesn't end with the given suffix throw an exception.
+     */
+    public static String replaceSuffix(String name, String oldSuffix, String newSuffix) {
```

Comment:
Nit: `name` should probably be `s` or something that makes it clear that it can be any string, not just a name.

Also, can we replace `CoreUtils.replaceSuffix` with this method and ensure we have unit tests for this new method?